cmd/watcher/main.go

```diff
@@ -76,12 +78,21 @@ func main() {
 	defer conn.Close()
 	results := v1alpha2pb.NewResultsClient(conn)
 
+	config := injection.ParseAndGetRESTConfigOrDie()
+
+	// Create kubernetes clientset
+	clientset, err := kubernetes.NewForConfig(config)
```

Comment:
I guess it would work, but only if we try to access it within the controller. In knative this is probably inserted in the shareadmin context by the `init()` functions of the injection kube clientset provided in the knative pkg.
Which is not a problem, in that case the client can be taken out this struct completely.
I will check on this.